docs/stopping_events/breakpoints.md

```diff
@@ -0,0 +1,109 @@
+---
+icon: material/sign-caution
+search:
+    boost: 4
+---
+# :material-sign-caution: Breakpoints
+Breakpoints are the killer feature of any debugger, the fundamental stopping event. They allow you to stop the execution of your code at a specific point and inspect the state of your program to find bugs or understand its design.
+
+!!! WARNING "Multithreading and Breakpoints"
+    **libdebug** breakpoints are shared across all threads. This means that if a breakpoint is hit in one thread, all threads will stop. You can use the [`hit_on()`](../debugging_flow/#hit-records) method of a breakpoint object to determine if a breakpoint was hit in a specific thread.
+
+A breakpoint can be inserted at any of two levels: *software* or *hardware*.
+
+### :octicons-code-24: Software Breakpoints
+Software breakpoints in the Linux kernel are implemented by patching the code in memory at runtime. The instruction at the chosen address is replaced with an interrupt instruction that is conventionally used for debugging. For example, in the `i386` and `AMD64` instruction sets, `int3` (0xCC) is reserved for this purpose.
+
+When the `int3` instruction is executed, the CPU raises a `SIGTRAP` signal, which is caught by the debugger. The debugger then stops the process and restores the original instruction to its rightful place.
+
+!!! INFO "Pros and Cons of Software Breakpoints"
+    Software breakpoints are unlimited, but they can break when the program uses self-modifying code. This is because the patched code could be overwritten by the program.
```

Comment:
They are also slower, most important IMO

Reply:
Addressed in [9e4a7f8](https://github.com/libdebug/libdebug/pull/129/commits/9e4a7f831a709f7ca168593df7deb81678b86ed5)